serpentTools/tests/test_ResultsReader.py

```diff
@@ -0,0 +1,233 @@
+                                   expectedKeff)
+        numpy.testing.assert_equal(self.reader.resdata['burnDays'],
+                                   expectedDays)
+
+    def test_univdata(self):
+        """Verify that the correct universes and values are present."""
+        expectedInfExp= ['infAbs', 'infCapt', 'infChid', 'infChip', 'infChit', 'infDiffcoef', 'infFiss', 'infFissFlx',
+                           'infFlx', 'infInvv', 'infKappa', 'infKinf', 'infMicroFlx', 'infNsf', 'infNubar', 'infRabsxs',
+                           'infRemxs', 'infS0', 'infS1', 'infS2', 'infS3', 'infS4', 'infS5', 'infS6', 'infS7',
+                           'infScatt0', 'infScatt1', 'infScatt2', 'infScatt3', 'infScatt4', 'infScatt5', 'infScatt6',
+                           'infScatt7', 'infTot', 'infTranspxs']
+        expectedMetaData = ['cmmDiffcoef', 'cmmDiffcoefX', 'cmmDiffcoefY', 'cmmDiffcoefZ', 'cmmTranspxs', 'cmmTranspxsX',
+                            'cmmTranspxsY', 'cmmTranspxsZ', 'macroE', 'macroNg', 'microE', 'microNg']
+        expectedinfValAbsSt0 = numpy.array([1.05040E-02, 1.23260E-01])
+        expectedinfValAbsSt1 = numpy.array([1.01031E-02, 5.62564E-02])
+        expectedinfUncAbsUn0 = numpy.array([0.00482, 0.00202])
+        expectedinfUncAbsUn1 = numpy.array([0.00156, 0.00071])
+        actualStates = set(self.reader.univdata.keys())
+        self.assertSetEqual(set(self.expectedStates), actualStates)   # check that all states are read
+        self.assertSetEqual(set(self.reader.univdata[self.expectedStates[1]].infExp.keys()),
+                            set(expectedInfExp))
+        self.assertSetEqual(set(self.reader.univdata[self.expectedStates[1]].metadata.keys()),
+                            set(expectedMetaData))
+        numpy.testing.assert_equal(self.reader.univdata[self.expectedStates[0]].infExp['infAbs'],
+                                   expectedinfValAbsSt0)
+        numpy.testing.assert_equal(self.reader.univdata[self.expectedStates[1]].infExp['infAbs'],
+                                   expectedinfValAbsSt1)
+        numpy.testing.assert_equal(self.reader.univdata[self.expectedStates[0]].infUnc['infAbs'],
+                                   expectedinfUncAbsUn0)
+        numpy.testing.assert_equal(self.reader.univdata[self.expectedStates[1]].infUnc['infAbs'],
+                                   expectedinfUncAbsUn1)
+
+
+class ResultsTesterFull(_ResultsTesterHelperFull):
+    """Class for testing the results reader."""
+
+    def test_raiseError(self):
+        """Verify that the reader raises an error for unexpected EOF"""
+        badFile = os.path.join(TEST_ROOT, 'bad_results_file.m')
+        with open(badFile, 'w') as badObj:
+            for _line in range(5):
+                badObj.write(str(_line))
+        badReader = ResultsReader(badFile)
+        with self.assertRaises(SerpentToolsException):
+            badReader.read()
+        os.remove(badFile)
+
+    def test_variables(self):
```

Comment:
But what variables are being tested? Change the name to reflect the scope and effect of this test

Reply:
def test_varsMatchSettings(self):